src/io/textfile_encoder.cc

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "singa/io/encoder.h"
+
+namespace singa {
+
+std::string TextEncoder::Encode(vector<Tensor>& data) {
+  CHECK_GE(data.size(), 1);
+  size_t size = data[0].Size();
+  const float* value = data[0].data<float>();
+  const char* ptr1 = reinterpret_cast<const char*>(value);
+  std::string d(ptr1, ptr1 + sizeof(float) * size);
+  if (data.size() == 2) {
+    const float label = (const float)data[1].data<int>()[0];
+    const char* ptr2 = reinterpret_cast<const char*>(&label);
```

Comment:
do not cast the float* to char*.
the encoded string is not readable. 
csv file is readable.